### PR TITLE
Execute dotnet run behind the hood in dotnet test

### PR DESCRIPTION
**Pull request by mariam-abdulla**

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-test/TestApplication.cs [hunk 1]
```diff
@@ -4,13 +4,15 @@
 using System.Collections.Concurrent;
 using System.Diagnostics;
 using System.IO.Pipes;
+using Microsoft.DotNet.Cli.commands.dotnet_test;
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestApplication.cs:7]: mmm what's this namespace?

File: src/Cli/dotnet/commands/dotnet-test/TestApplication.cs [hunk 2]
```diff
@@ -223,21 +224,63 @@ private void StoreOutputAndErrorData(Process process)
...
             return true;
         }
 
+        private string BuildArgsWithDotnetRun(BuiltInOptions builtInOptions)
+        {
+            StringBuilder builder = new();
+
+            builder.Append($"{CliConstants.DotnetRunCommand} {CliConstants.ProjectOptionKey} \"{_module.ProjectPath}\"");
+
+            if (builtInOptions.HasNoRestore || builtInOptions.HasNoBuild)
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestApplication.cs:240]: I think we should remove `builtInOptions.HasNoBuild` from this to be in line with user command line.

File: src/Cli/dotnet/commands/dotnet-test/TestApplicationActionQueue.cs [hunk 3]
```diff
@@ -23,7 +23,7 @@ public TestApplicationActionQueue(int dop, Func<TestApplication, Task<int>> acti
         public void Enqueue(TestApplication testApplication)
         {
             if (!_channel.Writer.TryWrite(testApplication))
-                throw new InvalidOperationException($"Failed to write to channel for test application: {testApplication.ModulePath}");
+                throw new InvalidOperationException($"Failed to write to channel for test application: {testApplication}");
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestApplicationActionQueue.cs:26]: this should be added to resources for localization

File: src/Cli/dotnet/commands/dotnet-test/TestModulesFilterHandler.cs [hunk 4]
```diff
@@ -50,7 +50,7 @@ public bool RunWithTestModulesFilter(ParseResult parseResult)
 
             foreach (string testModule in testModulePaths)
             {
-                var testApp = new TestApplication(testModule, _args);
+                var testApp = new TestApplication(new Module(testModule, null, null), _args);
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/commands/dotnet-test/TestModulesFilterHandler.cs:53]: I don't like to pass a null, can you expose a new ctor with only the testmodule?

**Review by MarcoRossignoli — APPROVED** [review]
LGTM some nits

File: src/Cli/dotnet/commands/dotnet-test/TestApplicationActionQueue.cs (hunk 3 above)
Comment (mariam-abdulla) [reply src/Cli/dotnet/commands/dotnet-test/TestApplicationActionQueue.cs:26]: I created a separate task for it.

**Review by MarcoRossignoli — APPROVED** [review]

**Comment by Evangelink** [comment]
Relates to #45927

**Pull request merged**